### PR TITLE
{CI} Add Test Rpm Package Mariner

### PR DESCRIPTION
**Pull request by bebound**
**Description**

Other builds have this test, add the same test for Mariner.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
CI

File: scripts/release/rpm/test_mariner_in_docker.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+
+# This script should be run in a centos7 docker.
+set -exv
+
+export USERNAME=azureuser
+
+tdnf --nogpgcheck install /mnt/rpm/$RPM_NAME -y
+
+tdnf install git gcc python3-devel python3-pip findutils -y
```
Comment (bebound) [inline scripts/release/rpm/test_mariner_in_docker.sh:10]: There are two differences with `test_rpm_in_docker.sh`
1. Mariner does not install `pip` when install `python3.9-devel`, it also does not have `python3.9-pip` package.
2. `dnf` becomes `tdnf`.

I don't want to add too many env variables, so I create a new script for Mariner.

Comment (jiasli) [inline scripts/release/rpm/test_mariner_in_docker.sh:10]: Can this script be reused from `test_rpm_in_docker.sh` by introducing some parameters?

Comment (bebound) [reply scripts/release/rpm/test_mariner_in_docker.sh:10]: There are some difference between Mariner and UBI, see https://github.com/Azure/azure-cli/pull/25747#discussion_r1132030505.

Using same script decreases the duplication code but increases the complexity in script.
We've used different dockerfile for Mariner and UBI. We can also use different test scripts.

**Review by jiasli — APPROVED** [review]

**Pull request merged**